src/phase-manager.ts

```diff
@@ -0,0 +1,527 @@
+import { FaintPhase } from "#app/phases/faint-phase";
+import { GameOverPhase } from "#app/phases/game-over-phase";
+import { LoginPhase } from "#app/phases/login-phase";
+import { MessagePhase } from "#app/phases/message-phase";
+import { MoveAnimPhase } from "#app/phases/move-anim-phase";
+import { MoveChargePhase } from "#app/phases/move-charge-phase";
+import { MovePhase } from "#app/phases/move-phase";
+import { NewBattlePhase } from "#app/phases/new-battle-phase";
+import { PokemonHealPhase } from "#app/phases/pokemon-heal-phase";
+import { SelectTargetPhase } from "#app/phases/select-target-phase";
+import { TitlePhase } from "#app/phases/title-phase";
+import { TurnInitPhase } from "#app/phases/turn-init-phase";
+import type { BattlerIndex } from "#enums/battler-index";
+import type { ChargeAnim } from "#enums/charge-anim";
+import type { MoveId } from "#enums/move-id";
+import type { PhaseId } from "#enums/phase-id";
+
+interface UseMoveInit {
+  pokemon: Pokemon;
+  targets: BattlerIndex[];
+  move: PokemonMove | MoveId;
+  /** Whether to add the {@linkcode MovePhase} to the front of the phase queue or defer it. */
+  when: "eager" | "defer" | "before" | "after";
+  phaseId?: PhaseId;
+  followUp?: boolean;
+  ignorePp?: boolean;
+  reflected?: boolean;
+  snatched?: boolean;
+}
+
+interface GameOverInit {
+  isVictory?: boolean;
+  clearPhaseQueue?: boolean;
+}
+
+interface ToTitleScreenInit {
+  /** Whether to clear the phase queue before adding the {@linkcode TitlePhase}. */
+  clearPhaseQueue?: boolean;
+  /** Whether to add the {@linkcode TitlePhase} to the front of the phase queue or defer it. */
+  eager?: boolean;
+}
+
+interface ToLoginScreenInit {
+  /** Whether to show text. @default true */
+  showText?: boolean;
+  /** Whether to add the {@linkcode LoginPhase} to the front of the phase queue or defer it. */
+  eager?: boolean;
+}
+
+interface PokemonFaintInit {
+  preventEndure?: boolean;
+  destinyTag?: DestinyBondTag | null;
+  grudgeTag?: GrudgeTag | null;
+  source?: Pokemon;
+}
+
+/**
+ * This is responsible for managing the game's {@linkcode Phase | phases}.
+ */
+export class PhaseManager {
+  /** dequeue/remove the first element to get the next phase */
+  private phaseQueue: Phase[] = [];
+  /** A temporary storage of what will be added to the front of {@linkcode phaseQueue} */
+  private phaseQueuePrepend: Phase[] = [];
+  /** overrides default of inserting phases to end of phaseQueuePrepend array, useful for inserting Phases "out of order" */
+  private phaseQueuePrependSpliceIndex: number = -1;
+  private conditionalQueue: Array<[() => boolean, Phase]> = [];
+  private nextCommandPhaseQueue: Phase[] = [];
+
+  private currentPhase: Phase | null = null;
+  private standbyPhase: Phase | null = null;
+
+  public getCurrentPhase<P extends Phase = Phase>(): P | null {
+    return this.currentPhase as P;
+  }
+
+  public getStandbyPhase(): Phase | null {
+    return this.standbyPhase;
```

Comment:
To match `getCurrentPhase()`:
```suggestion
  public getStandbyPhase<P extends Phase = Phase>(): P | null {
    return this.standbyPhase as P;
```